include/crm/common/ipc_attrd.h

```diff
@@ -47,6 +61,39 @@ typedef struct {
 int pcmk__attrd_api_query(pcmk_ipc_api_t *api, const char *host, const char *name,
                           const char *user_name, int options);
 
+/*!
+ * \internal
+ * \brief Tell pacemaker-attrd to update the CIB with current values
+ *
+ * \param[in] api           Connection to pacemaker-attrd
+ * \param[in] host          Affect only this host (or NULL for all hosts)
+ *
+ * \return Standard Pacemaker return code
+ */
+int pcmk__attrd_api_refresh(pcmk_ipc_api_t *api, const char *host);
+
+/*!
+ * \internal
+ * \brief Update an attribute's value, time to wait, or both
+ *
+ * \param[in] api           Connection to pacemaker-attrd
+ * \param[in] task          PCMK__ATTRD_CMD_UPDATE to update the value,
+ *                          PCMK__ATTRD_CMD_UPDATE_DELAY to update the dampening,
+ *                          or PCMK__ATTRD_CMD_UPDATE_BOTH to update both
+ * \param[in] host          Affect only this host (or NULL for all hosts)
+ * \param[in] name          Attribute name
+ * \param[in] value         The attribute's new value, or NULL to unset
+ * \param[in] dampen        The new time to wait value, or NULL to unset
+ * \param[in] user_name     ACL user to pass to pacemaker-attrd
+ * \param[in] options       Bitmask of pcmk__node_attr_opts
+ * \param[in] pattern       Is name a regex?
+ *
+ * \return Standard Pacemaker return code
+ */
+int pcmk__attrd_api_update(pcmk_ipc_api_t *api, const char *task, const char *host,
+                           const char *name, const char *value, const char *dampen,
+                           const char *user_name, int options, bool pattern);
+
```

Comment:
I'm guessing you plan on using a separate function to set multiple attributes at once?

Reply:
I haven't decided yet.  Using a single function would be nicer.  I think it's going to require server side changes, too.  That's what made me decide to send this PR out first and make sure I'm on the right track.

Reply:
It will definitely require server-side changes.

I could see something like this function except taking a list or hash table of name/value pairs. We'd have to decide whether to pass a single host for all or throw the host into the mix too. Keeping the single-attribute function could still be worthwhile for convenience, though it could become a wrapper or at least share most of the code.

Reply:
I think I'd like to address all the multiple attribute stuff with a follow up PR.  Probably what I'll do is change this function to take a GList of pcmk__attrd_query_pair_t (see the most current include/crm/common/ipc_attrd.h) structs.  That'll let me handle multiple hosts at a time, too.  Then I'll probably have a function that just does a single attribute so we don't have to make lists all the time.  And then the biggest part will be the server side changes and the command line option changes to let the tools do multiple attributes at once.